Sources/SpeziOnboarding/OnboardingFlow/OnboardingStepIdentifier.swift

```diff
@@ -13,38 +13,103 @@ import SwiftUI
 /// An `OnboardingStepIdentifier` serves as an abstraction of a step in the onboarding flow as outlined within the ``OnboardingStack``.
 /// 
 /// It contains both the identifier for an onboarding step (the view's type) as well as a flag that indicates if it's a custom onboarding step.
-struct OnboardingStepIdentifier: Hashable, Codable {
-    let custom: Bool
-    let identifierHash: Int
-
+struct OnboardingStepIdentifier {
+    /// The source of the `OnboardingStepIdentifier`'s identity
+    enum IdentifierKind {
+        /// The `OnboardingStepIdentifier` derives its identity from a `View`'s type and source location
+        case viewTypeAndSourceLoc
```

Comment:
This probably should have associated values for the viewType and source location?
For now, you also derive the view type of an `identifiable` View, do we actually need that? I think the manually specified identifier should be enough and always take precedence?
In general, moving away from the type of the View to be an identifier (by e.g. considering the source loc) is fantastic!

Reply:
i didn't do this because the presence of viewType and sourceLoc doesn't correlate with the kind of identifier: the viewType is always present (even if it isn't used as the step's primary source of itentity, we still want the ability to tell the path to just "go to the first step of type X"), and the sourceLoc is always present for steps defined via the `OnboardingFlowBuilder`.

It might be a good idea to rename the `IdentifierKind` to something else, to better indicate its purpose (defining how the `OnboardingStepIdentifier` gets its identity).